x-pack/platform/plugins/shared/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -143,103 +229,228 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
   }
 
   protected getResponseErrorMessage(error: AxiosError<{ error?: { message?: string } }>): string {
-    // handle known Azure error from early release, we can probably get rid of this eventually
     if (error.message === '404 Unrecognized request argument supplied: functions') {
-      // add information for known Azure error
       return `API Error: ${error.message}
         \n\nFunction support with Azure OpenAI API was added in 2023-07-01-preview. Update the API version of the Azure OpenAI connector in use
       `;
     }
     if (!error.response?.status) {
       return `Unexpected API Error: ${error.code ?? ''} - ${error.message ?? 'Unknown error'}`;
     }
-    // LM Studio returns error.response?.data?.error as string
     const errorMessage = error.response?.data?.error?.message ?? error.response?.data?.error;
     if (error.response.status === 401) {
       return `Unauthorized API Error${errorMessage ? ` - ${errorMessage}` : ''}`;
     }
     return `API Error: ${error.response?.statusText}${errorMessage ? ` - ${errorMessage}` : ''}`;
   }
+
   /**
    * responsible for making a POST request to the external API endpoint and returning the response data
    * @param body The stringified request body to be sent in the POST request.
    */
-
   public async runApi(
     { body, signal, timeout }: RunActionParams,
     connectorUsageCollector: ConnectorUsageCollector
   ): Promise<RunActionResponse> {
-    const sanitizedBody = sanitizeRequest(
-      this.provider,
-      this.url,
-      body,
-      ...('defaultModel' in this.config ? [this.config.defaultModel] : [])
-    );
-    const axiosOptions = getAxiosOptions(this.provider, this.key, false);
-    const response = await this.request(
-      {
-        url: this.url,
-        method: 'post',
-        responseSchema: RunActionResponseSchema,
-        data: sanitizedBody,
-        signal,
-        // give up to 2 minutes for response
-        timeout: timeout ?? DEFAULT_TIMEOUT_MS,
-        ...axiosOptions,
-        headers: {
-          ...this.headers,
-          ...axiosOptions.headers,
+    if (
+      this.provider === OpenAiProviderType.Other &&
+      (this.configAny.certificateFile ||
+        this.configAny.certificateData ||
+        this.configAny.privateKeyFile ||
+        this.configAny.privateKeyData)
+    ) {
+      const sanitizedBody = sanitizeRequest(
+        this.provider, this.url, body, ...('defaultModel' in this.configAny ? [this.configAny.defaultModel] : [])
+      );
+      const axiosOptions = getAxiosOptions(this.provider, this.key, false);
+      let sslOverrides: Record<string, any> | undefined;
+      if (
+        this.configAny.certificateFile ||
+        this.configAny.certificateData ||
+        this.configAny.privateKeyFile ||
+        this.configAny.privateKeyData
+      ) {
+        sslOverrides = getPKISSLOverrides(
```

Comment:
same with sslOverrides, if `this.hasPKI`, create a variable `this.sslOverrides` and refer to it throughout the class instead of calling multiple times